lib/Sema/TypeCheckAttr.cpp

```diff
@@ -7989,6 +7982,15 @@ void AttributeChecker::visitNonisolatedAttr(NonisolatedAttr *attr) {
     }
   }
 
+  // `nonisolated` on an actor declaration is contradictory.
+  if (auto *classDecl = dyn_cast<ClassDecl>(D);
+      classDecl && classDecl->isExplicitActor()) {
+    diagnoseAndRemoveAttr(attr, diag::invalid_decl_modifier, attr)
+        .warnUntilFutureSwiftVersion()
```

Comment:
can we be more aggressive than this? i don't think this construction is likely that common...

Reply:
I kicked off the source compat suite as well, we'd get some signal from it as well...

Generally it's better to not break source when it is low value like this

Reply:
```suggestion
```